org-roam.el

```diff
@@ -395,6 +395,51 @@ which takes as its argument an alist of path-completions.  See
   (interactive)
   (find-file org-roam-directory))
 
+;;;; org-roam-find-index
+(defcustom org-roam-index-file nil
+  "Path to the Org-roam index file.
+The path can be a string or a function.  If it is a string, it
+should be the path (absolute or relative to `org-roam-directory')
+to the index file.  If it is is a function, the function should
+return the path to the index file.  Otherwise, the index is
+assumed to be a note in `org-roam-directory' whose title is
+'Index'."
+  :type '(choice
+          (string :tag "Path to index" "%s")
+          (function :tag "Function to generate the path"))
+  :group 'org-roam)
+
+(defun org-roam--get-index-path ()
+  "Return the path to the index in `org-roam-directory'.
+The path to the index can be defined in `org-roam-index-file'.
+Otherwise, it is assumed to be a note in `org-roam-directory'
+whose title is 'Index'."
+  (let* ((index org-roam-index-file)
+         (path (pcase index
+                 ((pred functionp) (funcall index))
+                 ((or (pred stringp) 'nil) index)
+                 (wrong-type (signal 'wrong-type-argument
+                                     `((functionp stringp)
+                                       ,wrong-type))))))
+    (pcase index
+      ('nil (cdr (assoc "Index" (org-roam--get-title-path-completions))))
+      ((pred (f-relative-p)) (concat (file-truename org-roam-directory) path))
+      (_ index))))
+
+(defun org-roam-find-index ()
+  "Find the index file in `org-roam-directory'.
+The path to the index can be defined in `org-roam-index-file'.
+Otherwise, the function will look in your `org-roam-directory'
+for a note whose title is 'Index'.  If it does not exist, the
+command will offer you to create one."
+  (interactive)
+  (let ((index (org-roam--get-index-path)))
+    (if (and index
+             (file-exists-p index))
+        (find-file index)
+      (when (y-or-n-p "Index file does not exist.  Would you like to create it? ")
+        (org-roam-find-file "Index")))))
```

Comment:
it seems presumptuous to hardcode the file title here. Would this not ignore the `org-roam-index-file` variable?

Reply:
> it seems presumptuous to hardcode the file title here. Would this not ignore the `org-roam-index-file` variable?

The first arg of `org-roam-find-file` is for pre-filling the completion, so the user can still change it.  i can remove the `y-or-n-p` altogether if you want.

Reply:
Won't it then create an index file, at a path that's not the same as `org-roam-index-file` (if that's set)?

Reply:
It's because @progfolio suggested that we automatically-generate an index based on what's available in the db.  I've left the legacy `y-or-no-p` creation of file as a temporary measure.  I can look into better options if you want, or we can just `error` on `file-exists-p` in the meantime.

Reply:
Does 6d3cf8110d73e8f2a68591f8112598749e220532 look better to you?